src/ragas/metrics/_bleu_score.py

```diff
@@ -35,12 +31,13 @@ def init(self, run_config: RunConfig):
     async def _single_turn_ascore(
         self, sample: SingleTurnSample, callbacks: Callbacks
     ) -> float:
-        assert (
-            self.sentence_segmenter is not None
-        ), "Sentence segmenter is not initialized"
 
-        reference_sentences = self.sentence_segmenter.segment(sample.reference)
-        response_sentences = self.sentence_segmenter.segment(sample.response)
+        reference, response = sample.reference, sample.response
+        assert isinstance(reference, str), "Expecting a string"
+        assert isinstance(response, str), "Expecting a string"
```

Comment:
the fact that we have to use the assert here feels not good. The other problem is the error message, ideally it should tell them what to do about it, not sure if that is easy to define here

Reply:
I have edited the message to make it clear, but this is an API design problem. The metric expects valid response and reference but in SIngleTurnSample they are optional.